metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java

```diff
@@ -201,44 +236,82 @@ protected UpdateAspectResult ingestAspectToLocalDB(@Nonnull final Urn urn, @Nonn
       @Nonnull final AuditStamp auditStamp, @Nonnull final SystemMetadata providedSystemMetadata) {
 
     return _entityDao.runInTransactionWithRetry(() -> {
+      final String urnStr = urn.toString();
+      final EbeanAspectV2 latest = _entityDao.getLatestAspect(urnStr, aspectName);
+      long nextVersion = _entityDao.getNextVersion(urnStr, aspectName);
 
-      // 1. Fetch the latest existing version of the aspect.
-      final EbeanAspectV2 latest = _entityDao.getLatestAspect(urn.toString(), aspectName);
-      final EbeanAspectV2 keyAspect = _entityDao.getLatestAspect(urn.toString(), getKeyAspectName(urn));
+      return ingestAspectToLocalDBNoTransaction(urn, aspectName, updateLambda, auditStamp, providedSystemMetadata, latest, nextVersion);
+    }, DEFAULT_MAX_TRANSACTION_RETRY);
+  }
 
-      // 2. Compare the latest existing and new.
-      final RecordTemplate oldValue =
-          latest == null ? null : toAspectRecord(urn, aspectName, latest.getMetadata(), getEntityRegistry());
-      final RecordTemplate newValue = updateLambda.apply(Optional.ofNullable(oldValue));
+  @Override
+  @Nonnull
+  protected List<Pair<String, UpdateAspectResult>> ingestAspectsToLocalDB(@Nonnull final Urn urn,
+     @Nonnull List<Pair<String, RecordTemplate>> aspectRecordsToIngest,
+     @Nonnull final AuditStamp auditStamp, @Nonnull final SystemMetadata systemMetadata) {
 
-      // 3. If there is no difference between existing and new, we just update
-      // the lastObserved in system metadata. RunId should stay as the original runId
-      if (oldValue != null && DataTemplateUtil.areEqual(oldValue, newValue)) {
-        SystemMetadata latestSystemMetadata = EbeanUtils.parseSystemMetadata(latest.getSystemMetadata());
-        latestSystemMetadata.setLastObserved(providedSystemMetadata.getLastObserved());
+    return _entityDao.runInTransactionWithRetry(() -> {
 
-        latest.setSystemMetadata(RecordUtils.toJsonString(latestSystemMetadata));
+      final Set<String> aspectNames = aspectRecordsToIngest
+        .stream()
+        .map(Pair::getFirst)
+        .collect(Collectors.toSet());
+
+      Map<String, EbeanAspectV2> latestAspects = getLatestAspectEbeanForUrn(urn, aspectNames);
+      Map<String, Long> nextVersions = _entityDao.getNextVersions(urn.toString(), aspectNames);
+
+      List<Pair<String, UpdateAspectResult>> result = new ArrayList<>();
+      for (Pair<String, RecordTemplate> aspectRecord: aspectRecordsToIngest) {
```

Comment:
Are there tests for this class that should be updated?

Reply:
Nothing failed when I ran the gradle builds. Nothing failed on github. So I assumed tests have passed. Or are those not running with `./gradlew :metadata-service:war:build` and github?